packages/SwingSet/misc-tools/elapsed-per-method.py

```diff
@@ -35,26 +37,36 @@
                 msg = vd[2]
                 methargsCD = msg["methargs"]
                 result = msg["result"]
-                method = json.loads(methargsCD["body"])[0]
+                methodobj = json.loads(methargsCD["body"])[0]
+                if isinstance(methodobj, str):
+                    method = methodobj
+                elif isinstance(methodobj, dict) and methodobj["@qclass"] == "symbol":
+                    method = methodobj["name"]
+                else:
+                    raise ValueError("cannot determine method of %s" % json.dumps(methodobj))
                 start = when
             elif vd[0] == "notify":
                 method = "(notify)"
                 start = when
         if type == 'deliver-result':
             if start is not None:
                 elapsed = when - start
-                (old_count, old_elapsed) = deliveries[method]
+                computrons = data["dr"][2]["compute"]
+                (old_count, old_elapsed, old_computrons) = deliveries[method]
                 new_count = old_count + 1
                 new_elapsed = old_elapsed + elapsed
-                total += elapsed
-                deliveries[method] = (new_count, new_elapsed)
+                total_time += elapsed
+                new_computrons = old_computrons + computrons
+                total_computrons += computrons
+                deliveries[method] = (new_count, new_elapsed, new_computrons)
                 start = None
                 method = None
 
-print("| count| method                         | total time |  avg each |")
-print("| ---- | ------------------------------ | ---------- | --------- |")
+
+print("| count| method                         | total computrons | total time |  avg each |")
+print("| ---- | ------------------------------ | ---------------- | ---------- | --------- |")
 for method in sorted(deliveries, key=lambda method: deliveries[method][1], reverse=True):
-    (count, elapsed) = deliveries[method]
+    (count, elapsed, computrons) = deliveries[method]
     avg = elapsed / count
-    print("| %3dx | %30s |    %2.3f s |   %.3f s |" % (count, method, elapsed, avg))
-print("total: %2.3f s" % total)
+    print("| {:3d}x | {:>28} s | {:>16_d} |    {:2.3f} s |   {:.3f} s |".format(count, method, computrons, elapsed, avg))
```

Comment:
```suggestion
    f = "| {:5d} | {:>30} | {:>16_d} | {:>10.3f} | {:>8.3f} |"
    print(f.format(count, method, computrons, elapsed, avg))
```
(but really it'd be better to emit CSV and let something more general reformat to GFM)

Reply:
yeah, for tools like this, I always want human-readable output first, then maybe half the time I find a need to graph the results and then CSV is helpful, but it would be hard to do the eyeball analysis without the readably-formatted output, and I haven't yet hit the "three times" rule to build the more generic reformatter... what I think I'll do is, next time I need a graph from this, I'll add an `--as-csv` flag.